openfl-workspace/tf_cnn_histology/src/taskrunner.py

```diff
@@ -96,13 +96,30 @@ def create_model(self,
             metrics=[tf.keras.metrics.SparseCategoricalAccuracy()],
         )
 
-        self.tvars = model.layers
-        print(f'layer names: {[var.name for var in self.tvars]}')
+        return model
 
-        self.opt_vars = self.optimizer.variables()
-        print(f'optimizer vars: {self.opt_vars}')
+    def train_(self, batch_generator, metrics: list = None, **kwargs):
+        """
+        Train single epoch.
 
-        # Two opt_vars for one tvar: gradient and square sum for RMSprop.
-        self.fl_vars = self.tvars + self.opt_vars
+        Override this function for custom training.
 
-        return model
+        Args:
+            batch_generator (generator): Generator of training batches.
+                Each batch is a tuple of N train images and N train labels
+                where N is the batch size of the DataLoader of the current TaskRunner instance.
+            metrics (List[str]): A list of metric names to compute and save
+            **kwargs (dict): Additional keyword arguments
+
+        Returns:
+            list: Metric objects containing the computed metrics
+        """
+
+        history = self.model.fit(batch_generator,
+                                 verbose=1,
+                                 **kwargs)
+        results = []
+        for metric in metrics:
+            value = np.mean([history.history[metric]])
+            results.append(Metric(name=metric, value=np.array(value)))
```

Comment:
Out of curiosity, why do we take the mean value, as opposed to the latest one?

If we take accuracy for instance, wouldn't we be interested in the accuracy at the end of the training (which would certainly be higher than the mean value)?